task8/src/main/java/AdCollection/AdService.java

```diff
@@ -0,0 +1,158 @@
+                "https://raw.githubusercontent.com/annapetrachkova/educational-practice-/UI/UI/assets/makeup.jpg'");
+
+    }
+
+    public List<Ad> getPage(int skip, int top, Map<String, String> filterConfig) {
+        List<Ad> filteredAds = new ArrayList<Ad>();
+        for (Map.Entry pair : filterConfig.entrySet()) {
+            if (pair.getKey().equals("vendor")) {
+                list.stream()
+                        .filter(adv -> adv.getVendor().toLowerCase().contains(pair.getValue().toString().toLowerCase()))
+                        .forEach(filteredAds::add);
+            } else if (pair.getKey().equals("creationDate")) {
+                list.stream()
+                        .filter(adv -> adv.getCreatedAt().equals(pair.getValue()))
+                        .forEach(filteredAds::add);
+            }
+        }
+        if (filterConfig.size() == 0) {
+            filteredAds = new ArrayList<Ad>(list);
+        }
+        if (top > filteredAds.size()) {
+            top = filteredAds.size();
+        }
+        if (skip >= filteredAds.size()) {
+            skip = 0;
+        }
+        if (skip + top > filteredAds.size()) {
+            return filteredAds.subList(skip, filteredAds.size());
+        } else {
+            return filteredAds.subList(skip, skip + top);
+        }
+
+    }
+
+    public Ad getPost(int id) {
+        for (Ad item : list) {
+            if (item.getId() == id) {
+                return item;
+            }
+        }
+        return null;
+    }
+
+    public boolean validatePost(Ad adv) {
+        for (Ad item : list) {
+            if (item.getId() == adv.getId()) {
+                return false;
+            }
+        }
+        if (adv.getDescription() == null || adv.getDescription().length() > 200)
+            return false;
+        if (adv.getVendor() == null)
+            return false;
+
+        if (adv.getCreatedAt() == null)
+            return false;
+
+        return true;
+    }
+
+    public boolean addPost(Ad adv) {
+        if (validatePost(adv)) {
+            list.add(adv);
+            return true;
+        } else
+            return false;
+    }
+
+    public boolean editPost(int id, Ad filterConfig) {
+        Ad adv = getPost(id);
+        if (adv == null) {
+            return false;
+        }
+        if (filterConfig.getDescription() != null && filterConfig.getDescription().length() <= 200) {
+            adv.setDescription(filterConfig.getDescription());
+        }
+        if (filterConfig.getHashTags() != null) {
+            adv.setHashTags(filterConfig.getHashTags());
+        }
+        return true;
+    }
+
+    public boolean removePost(int id) {
+        Ad adv = getPost(id);
+        if (adv != null) {
+            list.remove(adv);
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    public String toJsonString(List<Ad> list) {
+        if (list.size() > 0) {
+            Gson gson = new Gson();
```

Comment:
I believe the `Gson` could serialize the whole collection at once, there's no need to serialize items one by one